docs/reference/sparql-reference.md

```diff
@@ -0,0 +1,254 @@
+# Reference templates for SPARQL queries 
+
+This document contains template SPARQL queries that can be adapted. 
+Comments are added in-code with `#` above each step to explain them so that queries can be spliced together
+
+## Checks/Report generation
+
+### All terms native to ontology 
+note: we assume that all native terms here have the same namespace - that of the ontology
+
+```SPARQL
+# select unique instances of the variable
+SELECT DISTINCT ?term
+WHERE {
+  # selecting where the variable term is either used as a subject or object 
+  { ?s1 ?p1 ?term . }
+  UNION
+  { ?term ?p2 ?o2 . }
+  # filtering out only terms that have the MONDO namespace (assumed to be native terms)
+  FILTER(isIRI(?term) && (STRSTARTS(str(?term), "http://purl.obolibrary.org/obo/MONDO_")))
+}
+```
+
+### Count class by prefixes
+Adaptable to counting queries
+
+```SPARQL
+# this query counts the number of classes you have with each prefix (eg number of MONDO terms, CL terms, etc.)
+
+# adding prefixes used
+prefix owl: <http://www.w3.org/2002/07/owl#>
+prefix obo: <http://purl.obolibrary.org/obo/>
+
+# selecting 2 variables, prefix and numberOfClasses, where number of classes is a count of distinct cls
+SELECT ?prefix (COUNT(DISTINCT ?cls) AS ?numberOfClasses) WHERE 
+{
+  # the variable cls is a class 
+  ?cls a owl:Class .
+  # removes any cases where the variable cls is blank
```

Comment:
@matentzn - dont quite understand this step, why is it needed?

Reply:
Which step, the "owl:Class" step? This is to ensure the query is not looking for object properties...

Reply:
oh no, the isblank step - just asked anita (we are in the office today) she said something to the extent of blank classes get "created" in triple stores, and that was to exclude. What I am a bit confused I guess is 1) why that is needed with the group by, and 2) why when i run this, there is still a blank row

Reply:
Ask us on Monday about blank classes, suffices to say: some nodes in RDF, like existential restriction, do not have an ID. So we want to ensure that these nodes are excluded from the query!